### ISSUE TITLE
OpenJDK 11 builds for Solaris (SPARC)?

### ISSUE THREAD

**Issue by karianna** [AdoptOpenJDK/openjdk-build#1279]
Is there any update on or any estimate when OpenJDK 11 builds will be available for Solaris (SPARC)?

_Originally posted by @sKiLLeFF in https://github.com/AdoptOpenJDK/TSC/issues/40#issuecomment-531114867_

**Comment by sKiLLeFF** [comment]
Is there any news to have this scheduled before the end of the year, for example by November?

**Comment by karianna** [comment]
@gdams have we got enough capacity for this?  I note that Solaris support is proposed to be dropped for Java 14+ (JEP 382)

**Comment by gdams** [comment]
I think that this should be closed until we know the future of OpenJDK on Solaris, We can re-consider at that point? @karianna ?

**Comment by karianna** [comment]
> I think that this should be closed until we know the future of OpenJDK on Solaris, We can re-consider at that point? @karianna ?

Does it just build-out of the box today?

**Comment by gdams** [comment]
> Does it just build-out of the box today?

It does but we don't currently have the hardware to build it.

**Comment by sKiLLeFF** [comment]
Just wondering, I noticed that as of right now OpenJDK 8 is still being built for Solaris (Sparc). Is the hardware of that build system compatible for building newer OpenJDK versions?

**Comment by karianna** [comment]
We need a Solaris box that also has Solaris 11 installed on it.

**Comment by 346L3** [comment]
Is there any way to set up a build slave for the build farm?
I'm also interested in more recent OpenJDK builds for Solaris and have Sparc and x86 servers running Solaris 10 & 11.
After a quick look, I could only find documentation on adding new variants to be built *on* the buildfarm, not provide farm systems themselves...

**Comment by karianna** [comment]
> Is there any way to set up a build slave for the build farm?
> I'm also interested in more recent OpenJDK builds for Solaris and have Sparc and x86 servers running Solaris 10 & 11.
> After a quick look, I could only find documentation on adding new variants to be built _on_ the buildfarm, not provide farm systems themselves...

@346L3 - please email infrastructureATadoptopenjdkDOTnet

**Comment by gdams** [comment]
@346L3 Are you still willing to connect some Solaris servers to our build farm? Please let me know if it's still an option. CC @sxa

**Comment by 346L3** [comment]
> @346L3 Are you still willing to connect some Solaris servers to our build farm? Please let me know if it's still an option. CC @sxa

Yes, absolutely. I just had a lot of stuff going on lately... Does one per type suffice?

**Comment by gdams** [comment]
@346L3 the more the better but we'd happily take anything that you have. Are you on our slack channel? If not would you join https://adoptium.net/slack and we can discuss over there?

**Comment by davidmoten** [comment]
Now that JDK 17 LTS is around I'm after a Solaris JDK 17  sparcv9 download. Still don't see anything after JDK 8 unfortunately. What's the status on the > JDK 8 builds?

**Comment by karianna** [comment]
Sadly that is not a supported version/platform combination

**Comment by sxa** [comment]
@davidmoten Solaris/SPARC was removed from the upstream openjdk codebase in late 2020 around the JDK15 timeframe, so we are unlikely to be able to build an official JDK17. (Also SPARC is an expensive platform for us to support)

**Comment by davidmoten** [comment]
> @davidmoten Solaris/SPARC was removed from the upstream openjdk codebase in late 2020 around the JDK15 timeframe, so we are unlikely to be able to build an official JDK17. (Also SPARC is an expensive platform for us to support)

Thanks for the reply. I did notice that Oracle JDK is now free again for JDK 17 and checked with them but they don't build Solaris sparc for JDK 15+ so I really should forget about it and hasten our migration off Solaris!

**Comment by sxa** [comment]
> Thanks for the reply. I did notice that Oracle JDK is now free again for JDK 17 and checked with them but they don't build Solaris sparc for JDK 15+ so I really should forget about it and hasten our migration off Solaris!

Yep it was [JEP 381](https://bugs.openjdk.org/browse/JDK-8241787) that stripped it out of the code base.
In the case of JDK11 there are a couple of other blockers for us building that one, but it is feasible which is the only reason this issue hasn't been finally closed :-)

**Comment by ptribble** [comment]
For people finding this, there are a couple of options.

You can get good builds of OpenJDK for Solaris on both SPARC and x86 from Liberica

https://bell-sw.com/pages/downloads/

These include JDK11, and also 12 and 13.

If you want something more recent, then you're into unsupported territory, although it's not impossible. I have patches to build current OpenJDK on x86 illumos, and OpenJDK up to 17 on Solaris SPARC.

https://github.com/ptribble/jdk-sunos-builder

and some binaries for Solaris SPARC up to JDK17 too

https://pkgs.tribblix.org/openjdk/sparc-solaris/
